File: Miscellaneous/Golang/237. Delete Node in a Linked List.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func main() {

	list := &ListNode{1, nil}
	list.Next = &ListNode{2, nil}
	list.Next.Next = &ListNode{3, nil}
	list.Next.Next.Next = &ListNode{4, nil}
	result := isPalindrome(list, 3)

	for result != nil {
		fmt.Println(result.Val)
		result = result.Next
	}
}

func isPalindrome(head *ListNode, num int) *ListNode {
	temp := head

	for temp != nil {
		if temp.Next.Val == num {
			temp.Next = temp.Next.Next
			return head
		} else {
			temp = temp.Next
		}
	}

	return head
}

File: Miscellaneous/Python/237. Delete Node in a Linked List.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def deleteNode(self, head, num):
        traveser = head

        while traveser is not None:
            if traveser.next.val == num:
                traveser.next = traveser.next.next
                return head
            else:
                traveser = traveser.next
       
        
        return head
        
obj = Solution()
list = ListNode(1)
list.next = ListNode(2)
list.next.next = ListNode(3)
list.next.next.next = ListNode(4)
ans = obj.deleteNode(list, 3)
while ans is not None:
    print(ans.val)
    ans = ans.next